src/interface/blas2_interface.hpp

```diff
@@ -598,6 +604,51 @@ typename sb_handle_t::event_t _tbmv_impl(sb_handle_t& sb_handle, index_t _N,
                              sb_handle.execute(assignOp, local_range));
 }
 
+template <uint32_t local_range_x, uint32_t local_range_y, uplo_type uplo,
+          transpose_type trn, diag_type diag, typename sb_handle_t,
+          typename index_t, typename container_t0, typename container_t1,
+          typename increment_t>
+typename sb_handle_t::event_t _tpmv_impl(sb_handle_t& sb_handle, index_t _N,
+                                         container_t0 _mA, container_t1 _vx,
+                                         increment_t _incx) {
+  static_assert(local_range_x % local_range_y == 0,
+                "Local y range needs to be a multiple of local x range.");
+
+  constexpr bool is_upper = (uplo == uplo_type::Upper);
+  constexpr bool is_transposed = (trn != transpose_type::Normal);
+  constexpr bool is_unit = (diag == diag_type::Unit);
+
+  constexpr index_t one = 1;
+
+  index_t vector_size = _N;
+  index_t matrix_size = ((_N + 1) * _N) / 2;
+
+  auto res_buffer =
+      blas::make_sycl_iterator_buffer<typename container_t0::scalar_t>(
+          vector_size);
+
+  auto mA = make_matrix_view<col_major>(_mA, one, matrix_size, matrix_size);
+  auto vx = make_vector_view(_vx, _incx, vector_size);
+  auto vres = make_vector_view(res_buffer, one, vector_size);
+
+  typename container_t0::scalar_t unused;
+
+  auto tpmv =
+      make_xpmv<local_range_x, local_range_y, /*is_symmetric=*/false, is_upper,
+                is_transposed, is_unit>(unused, mA, vx, unused, vres);
+
+  const index_t loc_mem_leading_dim = local_range_x + 1;
+
+  auto tpmvEvent = sb_handle.execute(
```

Comment:
Can we avoid making multiple calls to `sb_handle.execute()` here? This would lead to 2 independent `queue.submit()` calls and will also create a dependency between the two kernels as well which will need to be captured for the `usm` case.

Reply:
Thanks your review, @muhammad-tanvir-1211.

Unfortunately, I cannot. Basically, I'm forced to have an assign operation for `tpmv` since the routine result is stored in the same vector that contain the routine input, i.e. `vx`. For such a reason, I need a temporary vector (`vres`) for storing the solution that will be assigned to the input/output vector (`vx`) passed as routine argument.